### ISSUE TITLE
Main menu grouping

### ISSUE THREAD

**Issue by irina060981** [alpheios-project/alignment-editor-new#86]
I think we may need to add some grouping or additional levels to the main menu now that there are so many buttons. there isn't much logical order to them. 

Options, Upload and Download should be differentiated from actions that one does to create or edit alignments. 

Undo, Redo and Add Target are actions that can be done while aligning, and should be grouped together

Align should be separated from the rest, and more clearly associate with the act of submitting new texts (while creating an alignment from scratch or adding a new one)

We might also want to consider standard icons for Options (Settings), Upload, Download, Undo and Redo

Maybe we need a new issue for all of this though.

_Originally posted by @balmas in https://github.com/alpheios-project/alignment-editor-new/pull/85#discussion_r535646946_

**Comment by monzug** [comment]
Agree. before reading this item, I thought about ALIGN button not being visible enough. you have to look for it as it blends in with the other buttons.  after all, it's the most important action that needs to be taken by user.
also, I was thinking of making some of the buttons invisible until they can be selected. a session should start with Options Upload and Align. then add Undo Redo Add target Clear text.

**Comment by monzug** [comment]
Suggestion for cleaning up the menu bar:
1) why do we have an upload and download button at the top and also an upload/download in origin and target text? when I use the upload button from top, it prompts for uploading a file in the origin text, so it does the upload in origin text. wouldn't be better to remove the two buttons from the menu bar and keep only in origin and target text? 
2) in PC and Mac, the download/upload buttons  (and show metadata when enabled)  in origin and target texts have the white background always, no matter if  it's  standard theme or V1 theme. see attachment

![Screen Shot 2020-12-30 at 3 21 17 PM](https://user-images.githubusercontent.com/41396793/103414376-b074b100-4b4b-11eb-86f2-5839181b1e23.png)

**Comment by monzug** [comment]
Actually, I just realized that the download in the menu bar does download both the origin and target text. How does the upload for both origin and target text work then?

**Comment by irina060981** [comment]
>why do we have an upload and download button at the top and also an upload/download in origin and target text? when I use the upload button from top, it prompts for uploading a file in the origin text, so it does the upload in origin text. wouldn't be better to remove the two buttons from the menu bar and keep only in origin and target text?

>Actually, I just realized that the download in the menu bar does download both the origin and target text. and so does the upload in menu bar with both origin and target text in (file .tsv). maybe we should have a different way to present it?

@monzug,  you are right, the upper (main menu) has buttons for 
download/upload the whole alignment
For now it uploads/downloads origin sourceText and all targetTexts (with parameters - language, direction, sourceType).
Later (after #6, #69, #70, #71; #62, #77 ) it would have more rich outputs/inputs.

And the download/upload inside each text-editor-single-block is only for upload/download source text inside current text-block.
It won't be expanded, I think.

>in PC and Mac, the download/upload buttons (and show metadata when enabled) in origin and target texts have the white background always, no matter if it's standard theme or V1 theme. see attachment

would check theme variables for text-editor-single-block menu

**Comment by irina060981** [comment]
>>in PC and Mac, the download/upload buttons (and show metadata when enabled) in origin and target texts have the white background always, no matter if it's standard theme or V1 theme. see attachment

>would check theme variables for text-editor-single-block menu

would be fixed in https://github.com/alpheios-project/alignment-editor-new/pull/112

**Comment by irina060981** [comment]
Button's colors are fixed for actions menu.

**Comment by monzug** [comment]
1) Could you please fix also the new Reset all button? it has a white background.
2) also, if I click on download (from menu bar) when no files have been uploaded yet, I get the following error: You should define origin and target texts first.  if I click on download (from origin or target text) when no file has been uploaded, I don't get any error. Should I?

**Comment by irina060981** [comment]
>Could you please fix also the new Reset all button? it has a white background.

Oh I missed it - thank you for the catch!

> also, if I click on download (from menu bar) when no files have been uploaded yet, I get the following error: You should define origin and target texts first. if I click on download (from origin or target text) when no file has been uploaded, I don't get any error. Should I?

There are some requirements for starting download.
The main requirement - that there were defined texts for all source instance - origin and target.
You could upload texts there - ot type manually.

If you have even one empty textarea - then download of alignment (from the top menu) won't be available.
If you download one text - from text-editor-single-block, then you should have text inside this textarea.

**Comment by monzug** [comment]
@irina060981 so, you are saying that its ok if the two download (from main bar and the one from origin/target) behave differently when no file has been uploaded yet, correct?

**Comment by balmas** [comment]
@monzug I would not worry too much about the Download button for now. We still need to spec out the full requirements for that. 

I'm also not sure at all that we need the Download button for the texts themselves, at least not in the initial release. This is something that might become clearer as we work through the token/segment editor features.

**Comment by irina060981** [comment]
>@irina060981 so, you are saying that it's ok if the two downloads (from main bar and the one from origin/target) behave differently when no file has been uploaded yet, correct?

@monzug , Monica, I didn't fully understand, what do you mean here?
Download does not work for an empty textarea both for main menu download, and local text menu.

The only difference, that download main menu checks all text instances, and download in a local text menu checks only the current text block and its textarea.

Do you mean this difference in behave?

**Comment by monzug** [comment]
yes, that in one case I got an error message (download from main menu with empty text) and no message with empty text  in download from origin or target. As Bridget said, we should not be bother by the download button for now. Thanks

**Comment by irina060981** [comment]
>yes, that in one case I got an error message (download from main menu with empty text) and no message with empty text in download from origin or target.

oh, I see your concerns now - thank you for describing :) I would check, why the error message is not published in the case

**Comment by irina060981** [comment]
>Could you please fix also the new Reset all button? it has a white background.

Fixed with the PR

>>yes, that in one case I got an error message (download from main menu with empty text) and no message with empty text in download from origin or target.

>oh, I see your concerns now - thank you for describing :) I would check, why the error message is not published in the case

Fixed with the PR

**Comment by irina060981** [comment]
It would be good to add a left slide menu (similiar to texts.alpheios.net)
Add 3 modes for - text editor, aignment group editor and tokens editor, 
they should work similiar to tabs on the screen and be placed inside main menu

**Comment by irina060981** [comment]
- added left slide menu and move all menu buttons to it
- created tabs for
   - options
   - source text editor
   - alignment groups editor
   - tokens editor
   
   
![image](https://user-images.githubusercontent.com/12377640/113396697-4506dd80-93df-11eb-8709-c8b4b633973e.png)

**Comment by monzug** [comment]
at first glance, I noticed that the uploaded file should be cleared after clear text. in previous builds, we had the following
![Screen Shot 2021-04-06 at 2 28 17 PM](https://user-images.githubusercontent.com/41396793/113711102-d2e70f00-96e4-11eb-8d68-425e456de730.png)

with the sliding menu, we still have the latest file saved. see screenshot
![Screen Shot 2021-04-06 at 2 28 01 PM](https://user-images.githubusercontent.com/41396793/113711223-fc079f80-96e4-11eb-9d88-46a3f28ba89f.png)

**Comment by monzug** [comment]
would it look nicer to have in bold the editor I am currently in? e.g. in the following screenshot, in bold the alignment groups editor.
![Screen Shot 2021-04-07 at 4 11 56 PM](https://user-images.githubusercontent.com/41396793/113881377-ac94a280-97bc-11eb-9c3a-4c7216c70149.png)

**Comment by monzug** [comment]
at main level, I would add Align and Upload. so we do not need to open the slide menu.

**Comment by irina060981** [comment]
>would it look nicer to have in bold the editor I am currently in? e.g. in the following screenshot, in bold the alignment groups editor.

added

![image](https://user-images.githubusercontent.com/12377640/113965650-21c1af80-9871-11eb-84d2-b9f164f59092.png)

**Comment by monzug** [comment]
Great. Thanks!!!!

**Comment by monzug** [comment]
I think this was done a couple of weeks ago, so I'll close it.
